### PR TITLE
Hide unchanged cells

### LINKED ISSUE #174152: Notebook diff - minimap or hide unchanged cells
When viewing git diff for large notebooks, it's impossible to know what changed, since there is no visual indication where a change occurred. One option to solve this is showing a diff minimap like when diffing normal files. Another option is a checkbox to hide unchanged cells, as in the following picture (from Deepnote):

image

### PR DESCRIPTION
**Pull request by DonJayamanne**
Fixes #174152 

Recording 2024-08-07 at 15 28 08

### CODE REVIEW

File: src/vs/workbench/contrib/notebook/browser/diff/notebookDiffEditor.ts [hunk 1]
```diff
@@ -601,11 +604,39 @@ export class NotebookTextDiffEditor extends EditorPane implements INotebookTextD
...
 
-	private _setViewModel(viewModels: DiffElementViewModelBase[]) {
+	private readonly viewModelStore = this._register(new DisposableStore());
+	private _setViewModel(viewModels: DiffElementCellViewModelBase[]) {
 		this._diffElementViewModels = viewModels;
-		this._list.splice(0, this._list.length, this._diffElementViewModels);
+		this.viewModelStore.clear();
+
+		const newViewModel: IDiffElementViewModelBase[] = [];
+		let placeholder: DiffElementPlaceholderViewModel | undefined = undefined;
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/browser/diff/notebookDiffEditor.ts:613]: This is hacky, will need to create a class around this to make this simpler,
but this was the POC, rest seems about the state i want it in, still need to polish and fix some parts

File: src/vs/workbench/contrib/notebook/browser/diff/notebookDiffEditor.ts [hunk 2]
```diff
@@ -1027,56 +1058,6 @@ export class NotebookTextDiffEditor extends EditorPane implements INotebookTextD
 		};
 	}
 
-	getCellOutputLayoutInfo(nestedCell: DiffNestedCellViewModel) {
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/browser/diff/notebookDiffEditor.ts:1030]: Dead code

File: src/vs/workbench/contrib/notebook/browser/diff/diffComponents.ts [hunk 3]
```diff
@@ -58,6 +60,37 @@ export function getOptimizedNestedCodeEditorWidgetOptions(): ICodeEditorWidgetOp
 	};
 }
 
+export class CellDiffPlaceholderElement extends Disposable {
```
Comment (DonJayamanne) [inline src/vs/workbench/contrib/notebook/browser/diff/diffComponents.ts:63]: Base view model has nothing to dow with cells.
Using this we can now have a view model for diffing of notebook level metadata as well

**Review by bpasero — DISMISSED** [review]

**Review by rebornix — APPROVED** [review]
Looks good to me!

**Pull request merged**